pkg/cri/sbserver/image_status.go

```diff
@@ -33,20 +35,25 @@ import (
 // TODO(random-liu): We should change CRI to distinguish image id and image spec. (See
 // kubernetes/kubernetes#46255)
 func (c *criService) ImageStatus(ctx context.Context, r *runtime.ImageStatusRequest) (*runtime.ImageStatusResponse, error) {
+	ctx, span := tracing.StartSpan(ctx, makeSpanName("ImageStatus"))
+	defer span.End()
 	image, err := c.localResolve(r.GetImage().GetImage())
 	if err != nil {
+		span.RecordError(err)
```

Comment:
This is a good example of why defer matters. We actually arent going to error from the api when it `IsNotFound` so we shouldnt process it then.

Reply:
I was actually looking at it a little differently. If the image reference does not exist, client would only see an empty image status response and no error. Looking at the trace can provide information about the underlying error which was encountered. This will be true even if I use defer, sicne only returned errors will be caught.